tinygrad/runtime/support/compiler_hip.py

```diff
@@ -56,12 +58,21 @@ def compile_hip(prg:str, arch="gfx1100", asm=False) -> bytes:
   check(comgr.amd_comgr_destroy_action_info(action_info))
   return ret
 
+def compile_llvm(prg:str, arch="gfx1100") -> bytes:
+  with tempfile.NamedTemporaryFile(delete=True) as f:
+    llvm_backend = LLVMCompiler("AMDGPU", gpu=arch)
+    relo = llvm_backend.compile(prg, load=False)
+    f.write(relo)
+    f.flush()
+    args = [f.name, "--no-undefined", "-shared", "-o", "-"]
+    obj = subprocess.check_output([get_lld_path(), *args])
```

Comment:
This subprocess is really slow, and uses rocm. Can you avoid it and load like we load from CPU LLVM?